Makefile

```diff
@@ -43,9 +43,19 @@ generate: smithy-generate gen-config-asserts gen-repo-mod-replace tidy-modules-.
 smithy-generate:
 	cd codegen && ./gradlew clean build -Plog-tests && ./gradlew clean
 
-smithy-build:
+smithy-build: gen-repo-mod-replace
 	cd codegen && ./gradlew clean build -Plog-tests
```

Comment:
will this not run the codegen twice?

Reply:
The new task, `smithy-build-%` will be evaluated independently of the `smithy-build` because of the trailing `-%` on the task name. Running `make smith-build` will build everything, whereas `make smithy-build-com.amazonaws.rds` will only build API service Ids that start with `com.amazonaws.rds`.

The `gen-repo-mod-replace` task will only modify the already generated API client's `go.mod`, adding local `replace` statements for module dependencies within the SDK repo. this task will only run the gradle codegen once.